beacon_node/beacon_chain/src/beacon_chain.rs

```diff
@@ -853,9 +853,9 @@ impl<T: BeaconChainTypes> BeaconChain<T> {
 
         if state.slot > slot {
             return Err(Error::CannotAttestToFutureState);
-        } else if state.current_epoch() + 1 < epoch {
+        } else if state.current_epoch() < epoch {
             let mut_state = state.to_mut();
-            while mut_state.current_epoch() + 1 < epoch {
+            while mut_state.current_epoch() < epoch {
```

Comment:
I think we should also change the `build_committee_cache` call below to `RelativeEpoch::Current`